File: app/logic/go/logic/colorname.go
package logic

type color struct {
	Name    string
	R, G, B uint8
}

func dist(x, x0 uint8) uint8 {
	if x > x0 {
		return x - x0
	}
	return x0 - x
}

func Name(r, g, b uint8) string {
	output := ""
	const maxDistance = 10
	maxDistanceTotal := uint8(3 * maxDistance)

	for _, color := range colorNameLookupTable {
		var totalDist uint8
		d := dist(color.R, r)
		if d > maxDistance {
			continue
		}
		totalDist += d

		d = dist(color.G, g)
		if d > maxDistance {
			continue
		}
		totalDist += d

		d = dist(color.B, b)
		if d > maxDistance {
			continue
		}
		totalDist += d

		if totalDist == 0 {
			return color.Name
		}
		if totalDist < maxDistanceTotal {
			output = color.Name
			maxDistanceTotal = totalDist
		}
	}

	return output
}

File: app/logic/js/logic/colorname.js
import {lT} from "./colornamelookuptable.js";

export function name(r, g, b) {
    let o = "",
        d0 = 50.;

    lT.some((c) => {
        const dR = c.r - r,
            dG = c.g - g,
            dB = c.b - b;

        if (dR === 0 && dG === 0 && dB === 0) {
            o = c.n;
            return true;
        }

        const d = Math.sqrt(dR * dR + dG * dG + dB * dB);
        if (d < d0) {
            o = c.n;
            d0 = d;
        }
    });

    return o;
}
